File: web-filter/src/test/java/com/example/demo/ApplicationTests.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.example.demo;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationContext;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
import org.springframework.test.web.reactive.server.WebTestClient;

/**
 * @author hantsy
 */
@SpringJUnitConfig(classes = Application.class)
@ActiveProfiles("test")
public class ApplicationTests {

  @Autowired
  ApplicationContext context;

  WebTestClient rest;

  @BeforeEach
  public void setup() {
    this.rest = WebTestClient
        .bindToApplicationContext(this.context)
        .configureClient()
        .build();
  }

  @Test
  public void getAllPostsWithoutUserQueryParams_shouldReturn401() throws Exception {
    this.rest
        .get()
        .uri("/posts")
        .exchange()
        .expectStatus().isUnauthorized();
  }

  @Test
  public void getAllPostsWithUserQueryParams_shouldBeOK() throws Exception {
    this.rest
        .get()
        .uri("/posts?user=user")
        .exchange()
        .expectStatus().isOk();
  }

}

File: kotlin-gradle/src/test/kotlin/com/example/demo/ApplicationTests.kt
package com.example.demo

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInfo


class ApplicationTests {

    @Test
    @DisplayName("My 1st JUnit 5 test! 😎")
    fun `my first Junit 5 test`(testInfo: TestInfo) {
        assertEquals(2, 1 + 1, "1 + 1 should equal 2")
        assertEquals("My 1st JUnit 5 test! 😎", testInfo.displayName) { "TestInfo is injected correctly" }
    }
}